src/Resources/contao/classes/Frontend.php

```diff
@@ -500,20 +500,38 @@ protected function jumpToOrReload($intId, $strParams=null, $strForceLang=null)
 	 * @return boolean
 	 *
 	 * @deprecated Deprecated since Contao 4.5, to be removed in Contao 5.0.
-	 *             Use Frontend::getAuthenticationStatus() instead.
+	 *             Use Symfony security instead.
 	 */
 	protected function getLoginStatus($strCookie)
 	{
-		@trigger_error('Using Frontend::getLoginStatus() has been deprecated and will no longer work in Contao 5.0. Use Frontend::getAuthenticationStatus() instead.', E_USER_DEPRECATED);
+		$objTokenChecker = \System::getContainer()->get('contao.security.token_checker');
+
+		if ($strCookie == 'BE_USER_AUTH' && $objTokenChecker->hasBackendUser())
+		{
+			// Disable the cache if a back end user is logged in
+			if (TL_MODE == 'FE')
+			{
+				$_SESSION['DISABLE_CACHE'] = true;
+
+				// Always return false if we are not in preview mode (show hidden elements)
+				if (!$objTokenChecker->showUnpublished())
```

Comment:
This is a perfect example why the method should be named `isPreviewMode()`. Because the comment says "return false if we are not in preview mode" and then the code would say the same.

Reply:
There is a distinction between *preview mode* (using the frontend preview with frames) and *showing unpublished items*.

Reply:
That's probably true, however, we have not been making this distinction so far. We should rename the `Model::isPreviewMode()` method, too, then.

Still you have to use one of the prefixes. Maybe `canShowUnpublished()`?

Reply:
hah, I didn't know about that `Model` method. I guess we can argue that the preview mode is only unpublished elements, otherwise there is no difference between preview frames and just opening the page itself. I have renamed the method in 51220735aaea2435787580da126a939637efdacd to also match the model method name.